frontend/src/app/shared/components/op-context-menu/handlers/op-settings-dropdown-menu.directive.ts

```diff
@@ -310,6 +313,20 @@ export class OpSettingsMenuDirective extends OpContextMenuTrigger {
         icon: 'icon-custom-fields',
         onClick: () => false,
       },
+      {
+        // Calendar sharing modal
+        hidden: !this.showCalendarSharingOption,
+        disabled: this.authorisationService.cannot('query', 'getIcalUrl'),
+        linkText: this.I18n.t('js.toolbar.settings.share_calendar'),
+        icon: 'icon-link', // TODO: find sharing icons
+        onClick: () => {
+          if (this.authorisationService.can('query', 'getIcalUrl')) {
+            this.opModalService.show(QueryGetIcalUrlModalComponent, this.injector);
+          }
+
+          return true;
+        },
+      },
```

Comment:
> In case the user has the "Share calendars" permission but lacks "Manage calendars"
> 
> <img alt="image" width="1229" src="https://user-images.githubusercontent.com/617519/227565310-52d3784d-cbab-46d1-9d50-d2e0985571e3.png">
> 
> the dropdown menu is not show at all. That will prevent use cases of subscribing to a publicly available calendar.

I stumbled upon a similar issue as noted here https://github.com/opf/openproject/blob/4272806191f4c6f73fe5984585039ee4d50449fc/modules/calendar/spec/features/calendar_sharing_spec.rb#L72

How should we overcome these dependencies? For me, it still feels right from a product perspective to show this option in this dropdown. But maybe we need a separate sharing button then...

Reply:
Hi @jjabari-op and @ulferts,

This has been updated in the work package. The label now for the roles is "Subscribe to iCalendars".